deta/service.py

```diff
@@ -60,31 +60,46 @@ def _request(
         # close connection if socket is closed
         # fix for a bug in lambda
         try:
-            if self.client and os.environ.get("DETA_RUNTIME") == "true" and self._is_socket_closed():
+            if (
+                self.client
+                and os.environ.get("DETA_RUNTIME") == "true"
+                and self._is_socket_closed()
+            ):
                 self.client.close()
         except:
             pass
 
         # send request
         body = json.dumps(data) if content_type == JSON_MIME else data
-        client = self.client or http.client.HTTPSConnection(
+        self.client = self.client or http.client.HTTPSConnection(
```

Comment:
if we re-assign `self.client` then in a subsequent call to `request` it will use the same connection regardless if the `keepalive` parameter was set or not when we init the service

Reply:
Got it will fix this 👍🏽